.github/workflows/on-push-app-extensions.yml

```diff
@@ -0,0 +1,92 @@
+name: Pipeline Contentful App Extensions
+
+on:
+  push:
+
+jobs:
+  check-updates:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/yldio/asap-hub/node-python-sq:f93db8bae8be7528565c2510dbc8e3eece97983d
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    outputs:
+      GIT_DIFF: ${{ steps.diff.outputs.diff }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Add safe directory
+        run: git config --system --add safe.directory /__w/asap-hub/asap-hub
+      - name: Check if some app has changed
+        uses: technote-space/get-diff-action@v6
+        with:
+          PATTERNS: |
+            packages/contentful-app-extensions/**/*.+(ts|tsx)
+      - id: diff
+        run: |
+          echo "::set-output name=diff::${{ env.GIT_DIFF }}"
+
+  test:
+    needs: [check-updates]
+    if: ${{ needs.check-updates.outputs.GIT_DIFF != '' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/yldio/asap-hub/node-python-sq:f93db8bae8be7528565c2510dbc8e3eece97983d
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Run contentful app extension tests
+        run: yarn workspaces foreach -j 2 --include "*/contentful-app-*" -vp run test:no-watch
+
+  build-and-deploy:
+    needs: [test]
+    if: ${{ needs.check-updates.outputs.GIT_DIFF != '' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/yldio/asap-hub/node-python-sq:f93db8bae8be7528565c2510dbc8e3eece97983d
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup
+        id: setup
+        uses: ./.github/actions/setup-environment
+        with:
+          environment-name: Branch
```

Comment:
The values are in `Base` so they are the same for every environment, I added one hardcoded because I didn't wanted to make a different file for branch or dev/prod. Is there a cleaner way or is this good enough? :)

Reply:
You could just hardcode this to prod or even a different filename specific for this deployment?

Reply:
I'm going to change to `Production` because I'll change this to run only on `master`